kubemarine/plugins/kubernetes_dashboard.py

```diff
@@ -11,13 +11,96 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from typing import List, Optional
 
-from kubemarine.core import summary
+from kubemarine.core import summary, utils
 from kubemarine.core.cluster import KubernetesCluster
+from kubemarine.plugins.manifest import Processor, EnrichmentFunction, Manifest
 
 
 def schedule_summary_report(cluster: KubernetesCluster):
     plugin_item = cluster.inventory['plugins']['kubernetes-dashboard']
     hostname = plugin_item['hostname']
     # Currently we declare that Dashboard UI is available only via HTTPS
     summary.schedule_report(cluster.context, summary.SummaryItem.DASHBOARD_URL, f'https://{hostname}')
+
+
+class DashboardManifestProcessor(Processor):
+    def __init__(self, cluster: KubernetesCluster, inventory: dict,
+                 original_yaml_path: Optional[str] = None, destination_name: Optional[str] = None):
+        plugin_name = 'kubernetes-dashboard'
+        version = inventory['plugins'][plugin_name]['version']
+        if original_yaml_path is None:
+            original_yaml_path = f"plugins/yaml/dashboard-{version}-original.yaml"
+        if destination_name is None:
+            destination_name = f"dashboard-{version}.yaml"
+        super().__init__(cluster, inventory, plugin_name, original_yaml_path, destination_name)
+
+    def get_known_objects(self) -> List[str]:
+        return [
+            "Namespace_kubernetes-dashboard",
+            "ServiceAccount_kubernetes-dashboard",
+            "Service_kubernetes-dashboard",
+            "Secret_kubernetes-dashboard-certs",
+            "Secret_kubernetes-dashboard-csrf",
+            "Secret_kubernetes-dashboard-key-holder",
+            "ConfigMap_kubernetes-dashboard-settings",
+            "Role_kubernetes-dashboard",
+            "ClusterRole_kubernetes-dashboard",
+            "RoleBinding_kubernetes-dashboard",
+            "ClusterRoleBinding_kubernetes-dashboard",
+            "Deployment_kubernetes-dashboard",
+            "Service_dashboard-metrics-scraper",
+            "Deployment_dashboard-metrics-scraper",
+        ]
+
+    def get_enrichment_functions(self) -> List[EnrichmentFunction]:
+        return [
+            self.enrich_namespace_kubernetes_dashboard,
+            self.enrich_deployment_kubernetes_dashboard,
+            self.enrich_deployment_dashboard_metrics_scraper,
+        ]
+
+    def enrich_namespace_kubernetes_dashboard(self, manifest: Manifest):
+        key = "Namespace_kubernetes-dashboard"
```

Comment:
Could you add descriptions for that enrichment methods like in calico processor?